File: solutions/N2_2022_F1_Chuva.cpp
// NEPS 2072 - Chuva (https://neps.academy/br/exercise/2072)
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);

    int N, S;
    cin >> N >> S;

    vector<int> xs(N + 1);

    for (int i = 1; i <= N; ++i)
        cin >> xs[i];

    vector<int> ps(N + 1, 0);

    for (int i = 1; i <= N; ++i)
        ps[i] = ps[i - 1] + xs[i]; 

    vector<int> hist(1'000'001, 0);
    hist[0] = 1;

    long long ans = 0;

    for (int b = 1; b <= N; ++b)
    {
        if (ps[b] >= S)
            ans += hist[ps[b] - S];

        hist[ps[b]]++;
    }

    cout << ans << '\n';

    return 0;
}

File: solutions/N2_2022_F1_Chuva.py
# NEPS 2072 - Chuva (https://neps.academy/br/exercise/2072)

N = int(input())
S = int(input())
xs = [int(x) for x in input().split()]

xs = [0] + xs
ps = [0] * (N + 1)

for i in range(1, N + 1):
    ps[i] = ps[i - 1] + xs[i]

hist = dict()
hist[0] = 1

ans = 0;

for b in range(1, N + 1):
    if ps[b] - S in hist:
        ans += hist[ps[b] - S]

    if ps[b] in hist:
        hist[ps[b]] += 1
    else:
        hist[ps[b]] = 1

print(ans)
